### ISSUE TITLE
AmplitudeEmbedding normalization not behaving as expected

### ISSUE THREAD

**Issue by co9olguy** [XanaduAI/pennylane#365]
This simple example:
```
import numpy as np
import pennylane as qml
from pennylane.templates.embeddings import AmplitudeEmbedding

dev = qml.device('default.qubit', wires=3)
@qml.qnode(dev)
def circuit(data):
    AmplitudeEmbedding(data, wires=[0, 1, 2])
    return qml.expval.PauliZ(0)

data = np.ones(shape=(8,)) / np.sqrt(8)
circuit(data)
```
gives the error 
```
TypeError: unsupported operand type(s) for +: 'Variable' and 'Variable'
```

This seems to be due to the normalization check within the definition of `AmplitudeEmbedding`.

As a further point, the normalization check is probably too strict. Instead of checking if the normalization is exactly not equal to one, we should allow for some small numerical tolerance by using np.allclose()

**Comment by mariaschuld** [comment]
I agree, ran into the problem of normalisation condition being too strict too. Will replace norm == 1 check with isclose check. 

With regards to the error, is it wanted behaviour that amplitude encoding encodes trainable parameters?

**Comment by josh146** [comment]
> With regards to the error, is it wanted behaviour that amplitude encoding encodes trainable parameters?

That's a good point. In my (limited) experience I would expect it to encode raw data only, and so would then have to pass it via a keyword argument to avoid the `Variable + Variable` error @co9olguy got.

**Comment by mariaschuld** [comment]
A use case where variables have to be encoded is if there is classical pre-processing that scales the inputs by trainable parameters. So @co9olguy is right, we need to consider Variables beeing fed in.

**Comment by mariaschuld** [comment]
I will work on it!

**Comment by AroosaIjaz** [comment]
@mariaschuld Let me know if I can help in any way.

**Comment by mariaschuld** [comment]
Fixed in PR #376
